cirq/protocols/json_serialization.py

```diff
@@ -662,7 +663,7 @@ def default(self, o):
 def read_json(
     file_or_fn: Union[None, IO, pathlib.Path, str] = None,
     *,
-    json_text: Optional[str] = None,
+    json_text: Optional[Union[str, bytes]] = None,
```

Comment:
why did you need this change? might be better to force the user to `decode` if they're using `json_text`

Reply:
Looks like I saw mypy complaining about `read_json(json_str, ...)` in `read_gzip` and for some reason deciding to change this signature instead of decoding in `read_gzip`. Switched to use decode.